### DISCUSSION TITLE
Unable to upload file due to "Invalid object name 'media'" error

### DISCUSSION THREAD

**Discussion by mburkeSencon** [spatie/laravel-medialibrary#3017] [Q&A]
This may be a bug, but I've not seen any references to this error so I'll write it like a bug report and it can be moved if required.

**Background**
I have upgraded a Laravel project from 6 through to 9, which also meant upgrading Spatie Activity Log and Media Library incrementally. The server runs a new version of Ubuntu (Ubuntu 22), PHP 8.1 and the database is MSSQL.

**What I did**
Part of testing involved uploading a file to part of the software. This software was initially written and developed in Laravel 6. I took two files (a 7MB PDF file, and a 2b text file) and tried to upload these individually to the system.

**What I expected to happen**
I expected the file to upload like it did before, pre-upgrade.

**What happened**
A whole world of pain and anguish descended upon me. 

I mentioned two files were used. All of this started with a 7MB PDF file. Using the 7MB file returned an initial error of:

> laravel.ERROR: The current request does not have a file in a key named `media`

Then performing: 
`dd($request->files->get('media')->getErrorMessage())`
within the controller responsible for uploading the file returned (initially) that the file had only been partially uploaded. That's all the error I could get at that time.

I since found elsewhere that using a smaller file could make a difference in terms of troubleshooting, so I then decided to upload a 2 byte file. This returned a different error altogether. 

> SQLSTATE[42S02]: [Microsoft][ODBC Driver 18 for SQL Server][SQL Server]Invalid object name 'media'.

The table exists.

It's at this point I'm stuck. I'm not sure if it's a bug or not, but I know for myself personally I've spent all day with this and haven't got anywhere with it.

**Comment by parallels999** [comment]
Answered https://github.com/spatie/laravel-medialibrary/issues/3024#issuecomment-1240529270

>Either way, unfortunately this sounds like a deep database adapter issue (that might have been introduced by upgrading Laravel), and not something related to this package.
